extensions/oplogPopulator/allocationStrategy/AllocationStrategy.js

```diff
@@ -37,6 +39,30 @@ class AllocationStrategy {
     get maximumBucketsPerConnector() {
         throw errors.NotImplemented;
     }
+
+    /**
+     * Getter for the pipeline factory
+     * @returns {PipelineFactory} pipeline factory
+     */
+    get pipelineFactory() {
+        return this._pipelineFactory;
+    }
+
+    /**
+     * Process an old connector configuration, and return
+     * the list of buckets if the bucket list is valid against
+     * the current pipeline factory.
+     * @param {Object} oldConfig old configuration
+     * @returns {string[] | null} old configuration if valid
+     */
+    getOldConnectorBucketList(oldConfig) {
+        const bucketList = this.pipelineFactory.extractBucketsFromConfig(oldConfig);
+        if (this.pipelineFactory.isValid(bucketList)) {
+            return bucketList;
+        }
+        return null;
+    }
```

Comment:
this function does not have to do with the allocation strategy : the "validation" could be moved in the pipeline factory

Reply:
i moved it to the pipeline factory